### DISCUSSION TITLE
Force login and private videos open to user groups

### DISCUSSION THREAD

**Discussion by netzih** [mediacms-io/mediacms#600] [Q&A]
Hi,
I set up a mediacms server with docker and I'd like to keep the videos from being displayed to the public.
I would like for users to be required to log in to be able to see any videos and upload videos.
Also (maybe should be a separate issue), is there a way to create user groups and set videos to be open to certain groups only and others won't have access to it? (like setting it private but open to a certain group).
Thanks.

**Comment by Mozitux** [comment]
Hi, for your first request, simply set this
_PORTAL_WORKFLOW = "**public**"_
to this : 
_PORTAL_WORKFLOW = "**private**"_
In **settings.py** file line 18 _(/home/mediacms.io/mediacms/cms/)_

**Comment by netzih** [comment]
Thanks. 
I have a docker install, is it the same?
Is it in deploy/docker/local_settings.py?

**Comment by Mozitux** [comment]
I don't use Docker, but the exact file to modify is "settings.py"  on my side.
It should be the same name for docker installation and single server, I think.
Anyway, the local_settings.py configuration does not match settings.py, but it is in the same folder

EDIT: The folder should contain the following files:
![folders content mediacms](https://user-images.githubusercontent.com/19860756/177856708-e48deba7-4bd0-4305-9051-40005c610ce2.PNG)

**Comment by netzih** [comment]
So I changed it to private and rebooted the docker images but I can still access the vidoes when not logged in.
(I know it detected the change because I changed the time zone int hat file as well but I did it wrong at first so it would not boot up, So I know it is reading the changes I made)

**Comment by illuminati05** [comment]
Hey guys i have the same problem like netzih in the first post. I want to share videos for all registered users. But i can only set private (only for me --> without tag search and categories listings) or public for all (also without login). It think there must be an option inside of the video settings for "registered users" (private, public, not listed, REGISTERED USERS)

**Comment by s0ftcorn** [comment]
I achieved this by setting:

```
USERS_CAN_SELF_REGISTER = False
GLOBAL_LOGIN_REQUIRED = True
```

That way i control the registration: by making accounts manually through admin interface and then setting their password through manage.py inside the container, since i dont want my users to need an email.
Anyone without an login just sees the login screen and more importantly cant see any videos.
